src/components/certificates/EditCertificate.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+import CertificateForm from './CertificateForm';
+import { CertificatesContext } from '../../contexts/certificatesContext';
+import { Certificate } from '../../types/types';
+
+const EditCertificate: React.FC = () => {
+  const { id } = useParams<{ id: string }>();
+  const { certificates, loading, error } = useContext(CertificatesContext)!;
```

Comment:
I like that you understand how context is working, but in this case there is not really not need for it as you don't have so much nested components. But for new keep it do not spend time to remove it.

Anyway in EditCertificate view you do not need all certificates and than filter array to find the one by id. 

Here you have ID of the certificate and you fetch the one from backend (now from DB)